src/components/standalone-link/docs.mdx

```diff
@@ -0,0 +1,133 @@
+---
+componentName: StandaloneLink
+peerComponents:
+  - SwingsetTestIcon
+---
+
+The `StandaloneLink` component should be used when a link is needed outside of a body of text. See more documentation [the HDS Product Figma](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/?node-id=2365%3A21590).
+
+Internally, this component uses two other components:
+
+- `MaybeInternalLink`: to automatically determine whether or not `next/link` should be used
+- `Text`: to ensure conformance font size and weight with the design system
+
+## Props
+
+🚨 Guidance on icon sizes to pass into the `icon` prop:
+
+- If `textSize` is being set to 100 or 200, then size 16 icons are recommended
+- If `textSize` is being set to 300, then size 24 icons are recommended
+
+| Prop name      | Description                                                                      | Required? | Default value | Type                            |
+| -------------- | -------------------------------------------------------------------------------- | --------- | ------------- | ------------------------------- |
+| `className`    | A string of one or more class names. Applied last to the rendered `<a>` element. | No        | undefined     | string                          |
+| `href`         | The destination of the link.                                                     | Yes       | N/A           | string                          |
+| `icon`         | An icon from `@hashicorp/flight-icons` to render.                                | Yes       | N/A           | ReactElement                    |
+| `iconPosition` | Where the icon should be rendered within the link. See examples below.           | Yes       | N/A           | Either: 'leading' or 'trailing' |
+| `text`         | The text rendered within the `<a>` element.                                      | Yes       | N/A           | string                          |
+| `textSize`     | The `size` passed to the inner `Text` component.                                 | No        | undefined     | TextProps[`size`]               |
```

Comment:
Because we don't have a programmatic way to render icons just with a prop name (we've explored our options on this already), this component was built to accept an icon component for `icon` instead of just the name. That means we're not able to control the size of the icon automatically since imported icons already have a size, so I've named this prop `textSize` since the size of the text is all it is able to affect. Since this is something that is very manual and depends on the developer to choose the correct icon size, I've listed that guidance at the top of the Props section in this doc.